src/irmin/store.ml

```diff
@@ -849,27 +857,67 @@ module Make (B : Backend.S) = struct
         if Tree.equal test v then set_tree_once root key ~new_tree ~current_tree
         else err_test current_tree
 
-  let test_and_set_tree ?(retries = 13) ?allow_empty ?parents ~info t k ~test
-      ~set =
+  let test_and_set_tree_return_commit ?(retries = 13) ?allow_empty ?parents
```

Comment:
I think it would be nice to not need these helper functions. If you reorder the function declarations, I think this can be the body of `test_set_and_get_tree` and then `test_and_set_tree` can use that and ignore the commit.

Reply:
Good point :)) Fixed in https://github.com/mirage/irmin/pull/2074/commits/76985368191a62f5f9d819d29ad1595e58376de0